### PR TITLE
Re-raise callback exceptions if asked to

### PR DESCRIPTION
**Pull request by antoinealb**
This commit fixes issue #45, by adding a flag to the spin method. It
allows the user to catch exceptions thrown from messages handler. This
is useful during debugging, so that thrown exceptions can be caught by
debuggers.

Not yet tested on a real project as I don't have hardware with me. Will report testing result ASAP.

### CODE REVIEW

**Comment by pavel-kirienko** [comment]
I noticed that having to carry this parameter around seems clumsy and redundant. Would it not be easier to supply it to the node constructor and make it an unchangeable private attribute?

**Comment by antoinealb** [comment]
I changed the API to make this a node constructor parameter & added some unit tests.

File: uavcan/node.py [hunk 1]
```diff
@@ -190,11 +190,7 @@ def remove_handlers(self, uavcan_type):
     def call_handlers(self, transfer):
         for uavcan_type, wrapper in self._handlers:
             if uavcan_type == get_uavcan_data_type(transfer.payload):
-                # noinspection PyBroadException
-                try:
-                    wrapper(transfer)
-                except Exception:
-                    logger.error('Transfer handler exception', exc_info=True)
+                wrapper(transfer)
```
Comment (pavel-kirienko) [inline uavcan/node.py]: This change doesn't look correct. With your new logic, the first failing handler will prevent the library from invoking other handlers, which defeats the purpose of catching the exception in the library. Unless I misunderstood your intent, the old logic should be restored, with a try/except in every iteration.

Comment (antoinealb) [reply uavcan/node.py]: Oh yeah I did not understand the original logic correctly. Pushed a fix.

**Comment by pavel-kirienko** [comment]
Travis failed on Python 2.7 because of the exception handling syntax. Should we just ignore that, considering that the support for Python 2.7 is to be removed soon?

**Comment by antoinealb** [comment]
I think it is just because python 2 is lacking `unittest.mock`, installing mock should fix the issue (pushed now).

**Comment by pavel-kirienko** [comment]
I can't see your latest commit and a new CI build hasn't been triggered, are you sure it got pushed successfully?

**Comment by antoinealb** [comment]
Sorry, I think I got distracted between sending my comment and pushing my code. It should be OK now.

**Pull request merged**